ci/xbuild/wheel.py

```diff
@@ -416,11 +416,38 @@ def _get_python_tag(self):
         minor = sys.version_info.minor
         return "cp" + str(major) + str(minor)
 
+
     def _get_abi_tag(self):
-        soabi = sysconfig.get_config_var("SOABI")
-        parts = soabi.split("-")
-        assert parts[0] == "cpython"
-        return "cp" + parts[1]
+        """
+        Return the ABI tag if available, otherwise just emulate it.
+        Adopted from pypa/wheel, see https://github.com/pypa/wheel/blob/a51977075740fda01b2c0e983a79bfe753567219/src/wheel/bdist_wheel.py#L67
+        """
+
+        def get_flag(var, fallback):
+            val = sysconfig.get_config_var(var)
+            if val is None:
+                self.log.report_abi_variable_missing(var)
+                return fallback
+            return val
+
+        soabi = sysconfig.get_config_var('SOABI')
+        if soabi:
+            parts = soabi.split("-")
+            assert parts[0] == "cpython"
+            abi = 'cp' + parts[1]
+        elif hasattr(sys, 'maxunicode'):
+            d = ''
+            m = ''
+            if get_flag('Py_DEBUG', hasattr(sys, 'gettotalrefcount')):
+                d = 'd'
```

Comment:
it's probably simpler to start with `abi = self._get_python_tag()`, and then do `abi += 'd'` and `abi += 'm'` as necessary. That way we don't introduce too many extra variables

Reply:
Sure, these are artifacts from the old code.